extensions/wikia/Recirculation/services/FandomDataService.class.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+class FandomDataService {
+	static protected $instance = null;
+
+	const PARSELY_API_BASE = 'https://api.parsely.com/v2/';
+	const PARSELY_API_LIMIT = 5;
+	const PARSELY_API_PAGE = 1;
+	const PARSELY_API_SORT = '_hits';
+
+	const MCACHE_VER = '1.0';
+
+	protected function __construct() {
+	}
+
+	/**
+	 * get class instance
+	 * @return FandomDataService
+	 */
+	static public function getInstance() {
+		if ( FandomDataService::$instance == null ) {
+			FandomDataService::$instance = new FandomDataService();
+		}
+		return FandomDataService::$instance;
+	}
+
+	/**
+	 * Get posts for a specific type. Uses cache if available
+	 * @param string $type
+	 * @return an array of posts
+	 */
+	public function getPosts( $type ) {
+		$memcKey = wfMemcKey(__METHOD__, $type, self::MCACHE_VER);
+
+		$data = WikiaDataAccess::cache(
+			$memcKey,
+			WikiaResponse::CACHE_STANDARD, // 24 hours
+			function() use ( $type ) {
+				return $this->apiRequest( $type );
+			}
+		);
+
+		return $data;
+	}
+
+	/**
+	 * Make an API request to parsely to gather posts
+	 * @param string $type
+	 * @return an array of posts
+	 */
+	private function apiRequest( $type ) {
+		switch ($type) {
+			case 'shares':
+				$endpoint = 'shares/posts';
```

Comment:
Those could be class consts, but I don't mind them like that (if only used once).